### PR TITLE
Landing Page Controller + Update Feed styling

### PR DESCRIPTION
**Pull request by jaredkhan**
Sorry for tangling these together. Will take care to not do this in future.

Here, we define a controller that manages the data for the landing page and sends it to view models. 
We also style the updates feed based on discussion in #32 

I think this is a pretty nice set up. I like the idea of having view models that match up with BEM blocks though I haven't gone fully that way yet. 

I really like the way we can separate out our components here. It could be useful to make some sandbox route in the dev environment e.g. `/sandbox/PostCard` that allows us to see components in isolation

### CODE REVIEW

File: Sources/HaCTML/Attributes.swift [hunk 1]
```diff
@@ -1,13 +1,22 @@
 private func stringAttribute(_ key: String) -> AttributeKey<String> {
-  return .init(key, apply: { .text($0) })
+  return AttributeKey(key, apply: { .text($0) })
```
Comment (jaredkhan) [inline Sources/HaCTML/Attributes.swift]: I kept stumbling on trying to read this `.init`

File: Sources/HaCTML/Attributes.swift [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 }
 
 private func numberAttribute(_ key: String) -> AttributeKey<Int> {
-  return .init(key, apply: { .text(String($0)) })
+  return AttributeKey(key, apply: { .text(String($0)) })
 }
 
-private func CSSAttribute(_ key: String) -> AttributeKey<[String: String]> {
-  return .init(key, apply: { .text($0.map({ "\($0): \($1);" }).joined(separator: " ")) })
+private func CSSAttribute(_ key: String) -> AttributeKey<[String: String?]> {
```
Comment (jaredkhan) [inline Sources/HaCTML/Attributes.swift]: 50/50 on whether this is a good idea. Fundamentally this attribute deals with Strings to Strings but having this value be optional made things easier at the point of use (for me)

File: Sources/HaCWebsiteLib/Controllers/LandingPageController.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  ]
+
+  static var handler: RouterHandler = { request, response, next in
+    defer {
+      next()
+    }
+    do {
+      try response.send(
+        LandingPage(updates: updates).node.render()
+      ).end()
```
Comment (jaredkhan) [inline Sources/HaCWebsiteLib/Controllers/LandingPageController.swift]: `end` can throw a Socket.Error
We should decide on a way to stop every controller from having to catch this

File: Sources/HaCTML/Attributes.swift (hunk 1 above)
Comment (Pinpickle) [reply Sources/HaCTML/Attributes.swift]: Good call, should we give the other attribute functions the same treatment?

File: Sources/HaCTML/Attributes.swift (hunk 2 above)
Comment (Pinpickle) [reply Sources/HaCTML/Attributes.swift]: I share your sentiment. This should be a concern if an optional value can sneak its way in there unintentionally. Maybe we can come up with a friendly way to say "yeah this is optional but I'm doing it on purpose", so the behaviour is explicit?

File: Sources/HaCWebsiteLib/Controllers/LandingPageController.swift (hunk 3 above)
Comment (Pinpickle) [reply Sources/HaCWebsiteLib/Controllers/LandingPageController.swift]: How about extending the response type to have a `safelyEnd` method that catches and logs this error? While we're at it we could overload `send` to allow a `Node` type (too magical?)

File: Sources/HaCTML/Attributes.swift (hunk 1 above)
Comment (Pinpickle) [reply Sources/HaCTML/Attributes.swift]: I spoke to soon, you did exactly that!

File: static/src/styles/landing/LandingUpdateFeed.styl [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
+  padding-left: grid(4);
 }
 
 .LandingUpdateFeed__item {
-  padding: grid(2);
-  flex: 1 1 auto;
-  width: grid(40);
+  padding: grid(1);
+  display: flex;
+  width: 25%;
```
Comment (Pinpickle) [inline static/src/styles/landing/LandingUpdateFeed.styl]: Check out those gutters!

File: Sources/HaCWebsiteLib/ViewModels/PostCard.swift [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    case hackathon = "Hackathon"
+  }
+
+  let title: String
+  let category: PostCategory
+  let description: String
+  let backgroundColor: String?
+  let imageURL: String?
+  
+  var node: Node {
```
Comment (Pinpickle) [inline Sources/HaCWebsiteLib/ViewModels/PostCard.swift]: This is awesome!

Comment (jaredkhan) [reply Sources/HaCWebsiteLib/ViewModels/PostCard.swift]: Once we have made a few more components and have played with different patterns we can think about things like `ViewModel` protocols etc if we want to

**Pull request merged**